src/transformStateWithClones.js

```diff
@@ -7,7 +7,44 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyOfState = { ...state };
+  const anwser = [];
+
+  for (const actionObjects of actions) {
+    for (const action in actionObjects) {
+      if (actionObjects[action] === 'addProperties') {
+        for (const data in actionObjects['extraData']) {
+          copyOfState[data] = actionObjects['extraData'][data];
+        }
+
+        const copyBeforePushing = { ...copyOfState };
+
+        anwser.push(copyBeforePushing);
```

Comment:
```suggestion
        anwser.push(copyBeforePushing);
```
Щоб не писати це тричі, можна пушити в масив тільки один раз - після всіх перевірок на кожнійі ітерації

Reply:
Все змінив, окрім повторення, з того що я зрозумів з відео уроків, створюючи деструктирізацію тільки раз я створюю тільки одне посилання, тобто я буду пушити три рази той самий об'єкт, тому треба створити для кожного кейса окремий об'єкт, щоб було видно різницю після кожного пуша, я пробував багато різних варіантів і тільки цей спрацював.